setuptools/discovery.py

```diff
@@ -203,7 +223,16 @@ class FlatLayoutPackageFinder(PEP420PackageFinder):
     DEFAULT_EXCLUDE = tuple(chain_iter((p, f"{p}.*") for p in _EXCLUDE))
     """Reserved package names"""
 
-    _looks_like_package = staticmethod(_valid_name)
+    @staticmethod
+    def _looks_like_package(path: _Path, package_name: str) -> bool:
+        names = package_name.split('.')
+        return bool(
+            names and (
+                # Consider PEP 561
+                (names[0].isidentifier() or names[0].endswith("-stubs"))
+                and all(name.isidentifier() for name in names[1:])
+            )
+        )
```

Comment:
why the `bool( EXPR )`  ?

Reply:
Hi @gst thank you very much for having a look on the PR.

Locally I am using mypy to typecheck this module and try to catch any potential bug via static analysis. It turns out that the way Python works a logical expression does not necessarily returns a boolean value. Due to the short-circuit mechanisms it can return anything. In the case, without the `bool()`, the type of this expression is `Union[List[str], bool]`.

In practice, this does not really matters, the code works with or without `bool()` in the same way. But, if I leave the `bool()` there it means that I can keep relying on `mypy` to catch other bugs.

(In the future, I plan to suggest for setuptools to start incrementally adopting type hints and checks, in a module by module basis. `setuptools.discovery` would be ready for that).

Reply:
ok I understand. maybe a little comment to say about that before the return ?

Reply:
I had a second look at the expression.

I think I can safely skip the check for `names`.

This will simplify the logic expression and make the type converge to `bool` without the need for explicit cast.

I will submit a commit shortly.